### PR TITLE
fix(core): prevent graph output from being truncated

### LINKED ISSUE #18689: Get `nx graph` data via stdout
- [x] I'd be willing to implement this feature ([contributing guide](https://github.com/nrwl/nx/blob/master/CONTRIBUTING.md))

## Description
Add a mechanism to get project graph data out of nx in a manner useful for scripts.

## Motivation
`nx graph` by itself spins up a web server to serve an interactive graph, which is fine for humans but unusable for scripting. Sometimes it would be nice to programmatically get (for example) a list of the internal projects that a particular project depends on.

## Suggested Implementation
A `--stdout` option (or something similar) for the `nx graph` command.

## Alternate Implementations
There is a workaround, though it's a bit convoluted; `nx graph` already can output JSON if you give it a `--file=<something>.json` option (presumably this looks at the extension, since you can also do `--file=<something>.html` and get an HTML page). Thus, with some tempfile shenanigans, you can get a one-liner that does the same kind of thing, something like

```sh
$TMPDIR=$(mktemp -d) nx graph --file=$TMPDIR/graph.json && \
  jq -r ".graph.dependencies" $TMPDIR/graph.json
```

but that's ugly, and I'm honestly not sure how portable it is (my original attempts used other options of `mktemp` which aren't actually options in MacOS's version of the program).

Since there's already the mechanism to output JSON, this should be a fairly simple addition. But maybe there _is_ already a way (aside from my hack) and I just couldn't find it in the docs.

### PR DESCRIPTION
**Pull request by AgentEnder**
## Current Behavior
--graph stdout and --file stdout aren't documented. When printing graph to stdout, its possible for some output to get missed.

## Expected Behavior
Existing flags are documented, and `--print` is more discoverable. We await stdout finishing before exiting the program.

## Related Issue(s)

Fixes #18689

### CODE REVIEW

**Review by isaacplmann — APPROVED** [review]

File: docs/generated/cli/affected.md [hunk 1]
```diff
@@ -65,6 +65,24 @@ Use the currently executing project name in your command.:
  nx affected -t build --tag=$NX_TASK_TARGET_PROJECT:latest
 ```
 
+Preview the task graph that Nx would run inside a webview.:
```
Comment (FrozenPandaz) [inline docs/generated/cli/affected.md:68]:
```suggestion
Preview the task graph that Nx would run inside a webview:
```

Please fix this everywhere

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]

File: docs/generated/cli/show.md [hunk 2]
```diff
@@ -53,13 +53,13 @@ Show affected projects in the workspace, excluding end-to-end projects:
  nx show projects --affected --exclude=*-e2e
 ```
 
-Show detailed information about "my-app" in a json format.:
+Show detailed information about "my-app" in a json format:
 
 ```shell
  nx show project my-app
```
Comment (FrozenPandaz) [inline docs/generated/cli/show.md:59]: Please make sure this is correct..

File: docs/generated/cli/graph.md [hunk 3]
```diff
@@ -35,6 +35,18 @@ Generate a static website with project graph into an html file, accompanied by a
  nx graph --file=output.html
 ```
 
+Print the project graph as JSON to the console:
+
+```shell
+ nx graph --file=stdout
```
Comment (FrozenPandaz) [inline docs/generated/cli/graph.md:41]: Only show `--print`

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]

File: docs/generated/cli/graph.md [hunk 4]
```diff
@@ -89,7 +95,7 @@ Exclude certain projects from being processed
 
 Type: `string`
 
-Output file (e.g. --file=output.json or --file=dep-graph.html)
+Output file (e.g. --file=output.json or --file=dep-graph.html). Provide --file=stdout to print the project graph in the terminal.
```
Comment (FrozenPandaz) [inline docs/generated/cli/graph.md:98]: Let's not recommend this.

File: docs/generated/cli/graph.md [hunk 5]
```diff
@@ -141,6 +147,12 @@ Type: `number`
 
 Bind the project graph server to a specific port.
 
+### print
+
+Type: `boolean`
+
+Print the project graph in the terminal. (equivalent to --file=stdout)
```
Comment (FrozenPandaz) [inline docs/generated/cli/graph.md:154]: I don't think we have to mention that it's equivalent?

**Review by FrozenPandaz — CHANGES_REQUESTED** [review]

**Review by FrozenPandaz — APPROVED** [review]

**Pull request merged**